### ISSUE TITLE
Bug - Menus (dropdown, menutoggle, etc.) - first menu item in a dropdown has a highlight by default (and it shouldn't)

### ISSUE THREAD

**Issue by garrett** [patternfly/patternfly-react#10393]
**Describe the problem**
This is an issue with 
- [x] Patternfly 5  

[?] Patternfly 6  

Please give a clear and concise description of the problem. Which components are affected?

When clicking on a dropdown or menutoggle, the first item in a menu is already selected.

**How do you reproduce the problem?**

Open any dropdown, menutoggle, or other menu-based component and look at the first menu item. It looks like it's being hovered, but it isn't. This affects all the new menus on the PatternFly website as well.

**Expected behavior**

An item that isn't hovered or focused shouldn't look hovered or focused.

**Is this issue blocking you?**

We had this as part of a blocker for our release, as it's a regression from the previous (now deprecated) dropdowns and we recently ported to the new ones.

**Screenshots**

From PatternFly's website:

![image](https://github.com/patternfly/patternfly/assets/10246/e569009d-88ec-4d34-a52c-48a312063f4d)

![image](https://github.com/patternfly/patternfly/assets/10246/4b76e2b8-1dba-484a-ab77-708ab3259678)

![image](https://github.com/patternfly/patternfly/assets/10246/e754d886-1063-40db-88b6-aae253c3143a)

![image](https://github.com/patternfly/patternfly/assets/10246/2300d65e-160e-4f04-9904-0d4c5b1f6193)

![image](https://github.com/patternfly/patternfly/assets/10246/be56ea3d-4342-4770-9daa-f502e329ab31)

This affects Chrome too:

![image](https://github.com/patternfly/patternfly/assets/10246/2b5c79bb-7d9c-4bb9-9109-6ed2d058baf7)

And also WebKit:

![image](https://github.com/patternfly/patternfly/assets/10246/0a9405cf-9ec4-4a35-a50a-2b03c23e01b0)

**What is your environment?**
 - OS: Fedora Linux 40
 - Browser: Firefox 125.0.1, Chromium 124.0.6367.60, GNOME Web 46 (WebKitGTK 2.44.1, basically Safari 17 on Linux)

**What is your product and what release date are you targeting?**

Cockpit. Yesterday was the release. We're unfortunately now shipping with this bug, as we cannot indefinitely postpone the release to wait on a fix in a PatternFly release. :frowning_face:

**Comment by garrett** [comment]
Any news on this being fixed?

We still see this all the time in Cockpit, and would love for this regression to be fixed... and in PF5 (where it was introduced in April), not just PF6.

See this dropdown where the first one is selected and the one at the bottom has a highlight:

![image](https://github.com/patternfly/patternfly-react/assets/10246/e1d197ac-310c-49ca-bd17-7339a07d06e1)

It's really quite bad and confusing to have the same focus and hover stylings and have two items selected at the same time.

**Comment by thatblindgeye** [comment]
@garrett we currently have https://github.com/patternfly/patternfly-react/pull/10642 open for v5 that will allow disabling the auto-focus, but will require manual logic to focus an item (which will be needed when the dropdown is opened via keyboard at the very least). This would more help resolve scenarios where a dropdown may have something other than dropdown items (e.g. an input before the dropdown items) that should receive focus first. The issue with the hover/focus styling being the same may not be something we land for v5 since it's a topic that stretches beyond just Dropdown styling, though.

We also have https://github.com/patternfly/patternfly-react/issues/10543 for a post-v6 spike to investigate a more robust/customizable way to handle the autofocus, though again that wouldn't address the styling being the same for hover/focus.

**Comment by garrett** [comment]
Thanks for the response!

> The issue with the hover/focus styling being the same may not be something we land for v5

It's a very visible, confusing _regression_ that happened in a v5 point release. Projects cannot simply update to v6, as it introduces all kinds of breakages, including a radically different style that has fonts that are illegibly small. It will break all visual tests for projects too, so we're all at risk of having bugs slip through uncaught. (There will simply be way too many screens to compare to catch everything, without investing tons of time and hoping that everything is caught and addressed, including combinations of widgets the PatternFly team doesn't test together.)

Back to the issue filed here: This isn't one of the many bugs that can be papered over with local CSS overrides — it requires fixes in JavaScript, which means that all of the projects using PatternFly are depending on a fix from PatternFly directly.

This really is something that should be addressed in v5.

**Comment by thatblindgeye** [comment]
Does https://github.com/patternfly/patternfly-react/pull/10642 aid in addressing this for v5? The PR adds a prop to prevent that auto-focus behavior. Or is the ask more to only prevent that auto-focus behavior when using a mouse to click open the Dropdown (matching the behavior previously seen in Dropdown)?

**Comment by garrett** [comment]
@thatblindgeye: Thanks for looking into this!

The proposed solution doesn't fix the issue; it just adds a workaround. The default behavior should be correct without having to do anything special. Relying on a workaround means developers need to know about the problem and add extra code, which isn't ideal. This bug needs a direct fix.

For menus and menu-based components:

- Items shouldn't stay highlighted indefinitely
- Two items shouldn't be highlighted simultaneously
- Hovering over items, including sub-components, should update the highlight accordingly

This issue seems to stem from a code change that prioritized keyboard interactions over mouse interactions. Keyboard and mouse actions should behave differently, and their visual cues for focus and highlight should also differ. (Currently, in PatternFly 5, they look the same. But this bug should be able to be fixed without changing the style.)

Fixing this in PatternFly 5 is quite imporant as software using this version will be in many enterprise products for around a decade. Upgrading to PatternFly 6 will require significant changes and moving away from deprecated components, so many projects will not do this in the immediately and will be stuck with PatternFly 5 (and this confusing issue, unless fixed) for a very long time.

**Comment by thatblindgeye** [comment]
The reason for this behavior is due to the change made for https://github.com/patternfly/patternfly-react/issues/9788, which we feel also helps alleviate an accessibility issue. While the original issue was more of a general ask for multi-device support and the intent of the PR was to prioritize combo mouse+keyboard interaction rather than solely one or the other (e.g. mouse click a menu toggle, but be able to immediately navigate a menu via keyboard arrow keys), in some testing we found that certain AT looked to benefit from this behavior as well; where voice input is used to command actions, stating "click"/"touch" on the toggle acted similarly to a native mouse click, and without auto-focusing the menu content it could be very tedious and require more-than-necessary effort to get into the menu/click a menu item.

While it's a bit confusing at first that the first item in a Dropdown menu has that focus styling and another item will have the hover styling when hovered via mouse, the behavior itself is intended. It's unfortunate that we're relying on native browser focus outlines which causes both a focused item and hovered item to have the same styling under certain circumstances, and that there's some odd browser behavior happening where:

- Focusing the menu toggle via keyboard and pressing Enter or Space to open the Dropdown will apply the browser-defined focus outline to the auto-focused item (this has been the behavior for keyboard interaction for a while)
- Focusing the menu toggle via keyboard and clicking via mouse to open the Dropdown will also apply the browser-defined focus outline
- However, clicking the menu toggle via mouse without it first being focused *won't* apply that browser-defined focus outline

It's worth noting that the behavior where 1 item can have hover style and another item can have focus styles predates this recent behavior, and I don't believe these components fully mimicked a native `select` styling behavior (which a native element looks to match the styling as you described in your bullet points above). The main difference really being that we autofocus on mouse click now. Though this behavior of auto-focusing on mouse click is newer and different behavior than previously, it was an intentional update due to the aforementioned consumer request + feeling it helps alleviate a11y.

Depending on bandwidth this may be something we could look into addressing for v5 after v6 is released, though @nicolethoen may have more insight on that once we get further to the v6 release.

**Comment by garrett** [comment]
Still not stale, still a regression, still not fixed in PF5 (where it was introduced in a minor version on an update) and even seems to be a problem in PF6 as well, according to the staging site.

As seen on https://staging-v6.patternfly.org/get-started/design/ — on any dropdown, even the one in the top-right:

![image](https://github.com/user-attachments/assets/1d7b64da-1887-4e71-849c-f87f35bb9bf1)
